server/controllers/index.js

```diff
@@ -15,5 +16,7 @@ router.get('/api/events', getAllEvents)
 router.get('/api/available-rooms', searchAvailableRooms)
 router.post('/api/login', login)
 router.post('/api/booking', addEvent)
+router.use(errors.serverErr)
+router.use(errors.notFound)
```

Comment:
does this actually work? I don't think this will ever get invoked. If you want to have a generic error handling in your application, you have to:
1. add an error handling function to the `app` (not the router!) right after you add the router
```js
app.use(router);
app.use(errorHandler);
```
2. The `errorHandler` should look something like this (taken from some interview assignment I did):
```js
function (err, req, res, next) {
    if (err instanceof NotFoundError) {
        console.debug(err);
        res.status(404).end();
    } else if (err instanceof BadRequestError || err instanceof SyntaxError) {
        console.debug(err);
        res.status(400).json({errMsg: err.message})
    } else {
        console.error(err);
        res.status(500).json({errMsg: 'Operation failed. See logs for more details'});
    }
}
```
The `NotFoundError`, `BadRequestError` and `SyntaxError` are my own custom errors I defined in my application.
3. For the error handler to get invoked, you have to use the `next` function in the controllers in case of error:
```js
router.post('/', function createOwner(req, res, next) {
    OwnerModel
        .create(req.body)
        .then(/* submit success resp */)
        .catch((err) => next(translateDbError(err)));
});
```
The `translateDbError` is a function that will produce one of the custom errors above

Reply:
482106
Thank you Janci, I think she deals with it as a frontend, it stills wrong and my fault to not note that,
We will change what you asked in other comments but still not using middleware, does that good? I mean not using middleware is bad or there is no one way?

Reply:
no that's fine, you can handle the error response directly in your controllers. Essentially the function from point 2 above can be called from the controller's `catch` and then you don't have to set it up through middleware. Which, since I don't see any other difference between the two approaches, might be even easier...